bin/tx-prover/build.rs

```diff
@@ -0,0 +1,96 @@
+use std::{
+    env, fs,
+    path::{Path, PathBuf},
+};
+
+use miette::IntoDiagnostic;
+use protox::prost::Message;
+
+/// Generates Rust protobuf bindings from .proto files in the root directory.
+///
+/// This is done only if BUILD_PROTO environment variable is set to `1` to avoid running the script
+/// on crates.io where repo-level .proto files are not available.
+fn main() -> miette::Result<()> {
+    println!("cargo::rerun-if-changed=../../proto");
+    println!("cargo::rerun-if-env-changed=BUILD_PROTO");
+
+    // Skip this build script in BUILD_PROTO environment variable is not set to `1`.
+    if env::var("BUILD_PROTO").unwrap_or("0".to_string()) == "0" {
+        return Ok(());
+    }
+
+    compile_tonic_server_proto()?;
+
+    Ok(())
+}
+
+fn compile_tonic_server_proto() -> miette::Result<()> {
+    let crate_root: PathBuf =
+        env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR should be set").into();
+    let dst_dir = crate_root.join("src").join("server").join("generated");
+
+    // Remove all existing files.
+    fs::remove_dir_all(&dst_dir).into_diagnostic()?;
+
+    fs::create_dir(&dst_dir).into_diagnostic()?;
+
+    // Compute the directory of the `proto` definitions
+    let cwd: PathBuf = env::current_dir().into_diagnostic()?;
+
+    let proto_dir: PathBuf = cwd.join("proto");
```

Comment:
Is there a reason to build the path like this rather than do it like we do for `dst_dir` on line 30 above?

Reply:
No, this is not needed. Im removing it.